src/SW_LIB.py

```diff
@@ -1756,7 +1756,7 @@ def GET_MEM_H_LOGIC_LOOKUP(parser_state):
                                 dim = dims[i]
                                 addr_t = (
                                     "uint"
-                                    + str(int(math.ceil(math.log(dim, 2))))
+                                    + str(int(math.ceil(math.log(dim, 2)))+1) #round up by 1
```

Comment:
Reasonable, but I prefer to keep the width exact as needed

For ex. you need two address bits to address four things
but the round up by one makes it 3 bits for dim=4

hmm maybe just need a simple 'its uint1 unless >0 then do math as is'

Reply:
There is another place this math occurs at  for sizing muxes in this same file

https://github.com/JulianKemmerer/PipelineC/blob/bf7e57f1a3a5be6cf3b9a034c2772b0482c1b79c/src/SW_LIB.py#L2060

mind changing there too?


And searching repo I found one more place
https://github.com/JulianKemmerer/PipelineC/blob/bf7e57f1a3a5be6cf3b9a034c2772b0482c1b79c/src/C_TO_LOGIC.py#L2307


could maybe make a common helper func in https://github.com/JulianKemmerer/PipelineC/blob/master/src/utilities.py
but not critical